### PR TITLE
feat: add GLOBAL_WORKING_DIR and GLOBAL_WORKING_PROCESS_DIR config parameteres

### PR DESCRIPTION
**Pull request by amadeusz-ds**
This PR introduces GLOBAL_WORKING_DIR and GLOBAL_WORKING_PROCESS_DIR controlling where temporary files are stored during partition flow, via tempfile.tempdir.

#### Edit:
Renamed prefixes from STORAGE_ to UNSTRUCTURED_CACHE_

#### Edit 2:
Renamed prefixes from UNSTRUCTURED_CACHE to GLOBAL_WORKING_DIR_

### CODE REVIEW

File: unstructured/partition/utils/config.py [hunk 1]
```diff
@@ -117,5 +140,28 @@ def PDF_ANNOTATION_THRESHOLD(self) -> float:
...
         return self._get_float("PDF_ANNOTATION_THRESHOLD", 0.9)
 
+    @property
+    def STORAGE_ENABLED(self) -> bool:
+        """Enable usage of STORAGE_DIR and STORAGE_TMPDIR."""
+        return self._get_bool("STORAGE_ENABLED", False)
+
+    @property
+    def STORAGE_DIR(self) -> str:
+        """Path to Unstructured storage directory."""
```
Comment (cragwolfe) [inline unstructured/partition/utils/config.py:150]: STORAGE_DIR is a misleading name, which has permanent or at least caching connotations. could this instead be TMP_STORAGE_DIR?

Comment (amadeusz-ds) [reply unstructured/partition/utils/config.py:149]: There are parameters `STORAGE_DIR` and `STORAGE_TMPDIR`

Comment (amadeusz-ds) [reply unstructured/partition/utils/config.py:149]: Maybe `UNSTRUCTURED_DIR` and `UNSTRUCTURED_TMPDIR` as those point to `~/.cache/unstructured` and `~/.cache/unstructured/tmp/{gid}` respectively

Comment (pawel-kmiecik) [reply unstructured/partition/utils/config.py]: or `UNSTRUCTURED_CACHE_DIR` and `UNSTRUCTURED_TMP_DIR`?

**Review by pawel-kmiecik — APPROVED** [review]
LGTM, just env names to be corrected.

File: unstructured/metrics/evaluate.py [hunk 2]
```diff
@@ -160,7 +160,6 @@ def _try_process_document(self, doc: Path) -> Optional[list]:
     @abstractmethod
     def _process_document(self, doc: Path) -> list:
         """Should return all metadata and metrics for a single document."""
-        pass
```
Comment (cragwolfe) [inline unstructured/metrics/evaluate.py:163]: why removed?

Comment (amadeusz-ds) [reply unstructured/metrics/evaluate.py:163]: It was removed by the linter

File: test_unstructured/partition/pdf_image/test_pdf_image_utils.py [hunk 3]
```diff
@@ -161,7 +163,12 @@ def test_save_elements_with_output_dir_path_none():
         )
 
         # Verify that the images are saved in the expected directory
-        expected_output_dir = os.path.join(tmpdir, "figures")
+        if storage_enabled:
```
Comment (micmarty-deepsense) [inline test_unstructured/partition/pdf_image/test_pdf_image_utils.py:166]: personally I'd prefer to see more usages of pathlib, but I believe it's not the mail goal of this PR, just a side note

**Review by cragwolfe — APPROVED** [review]

**Pull request merged**